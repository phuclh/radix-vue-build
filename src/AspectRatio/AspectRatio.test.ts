import { beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'
import { AspectRatio } from './'
import type { VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'

describe('given a default AspectRatio', async () => {
  let wrapper: VueWrapper<InstanceType<typeof AspectRatio>>

  beforeEach(() => {
    wrapper = mount(AspectRatio, {
      slots: {
        default: `<img
        class="h-full w-full object-cover"
        src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
        alt="Landscape photograph by Tobias Tullius"
      >`,
      },
    })
  })

  it('should pass axe accessibility tests', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  it('should render as snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
