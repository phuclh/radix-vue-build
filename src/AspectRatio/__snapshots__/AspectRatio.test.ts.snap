// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`given a default AspectRatio > should render as snapshot 1`] = `
"<div style="position: relative; width: 100%; padding-bottom: 100%;" data-radix-aspect-ratio-wrapper="">
  <div style="position: absolute; inset: 0px;"><img class="h-full w-full object-cover" src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&amp;dpr=2&amp;q=80" alt="Landscape photograph by Tobias Tullius"></div>
</div>"
`;
